addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -94,9 +94,14 @@ def _is_readonly(self):
         return False
 
     def _compute_readonly_timesheet(self):
-        readonly_timesheets = self.filtered(lambda timesheet: timesheet._is_readonly())
-        readonly_timesheets.readonly_timesheet = True
-        (self - readonly_timesheets).readonly_timesheet = False
+        # Since the mrp_module gives write access to portal user on timesheet, we check that the user is an internal one before giving the write access.
+        # It is not supposed to be needed, since portal user are not supposed to have access to the views using this field, but better be safe than sorry
```

Comment:
I'm confused :thinking: So is this necessary to impose access rights or are we doing it to make sure that access rights are reliably enforced in the future ? Basically, is this required or we're doing it to be sure (because why wouldn't we be sure?) ?

Reply:
well, this is actually a leftover of my first try to fix the issue. I planned to use it to force the fields to be in readonly for the portal user (which worked), but was not enough, since the field in readonly were still openable. And that is something that i could not fix, and so i was forced to use different view instead. So, with the fix as it is, the portal user should not be able to access a view which will require this computed field. But I thougth that it was a good safety net to leave it here. In case we left somewhere a way for the user to reach them (or add one later on), he still won't be able to modify timesheets. But i dont have a strong opinion about it, if you think its overkill, i dont mind removing it.